src/server/src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -66,75 +86,114 @@ private GossipInfo parseEndpointStatesString(String sourceNode, String allEndpoi
       Optional<String> releaseVersion = Optional.absent();
       Optional<String> tokens = Optional.absent();
       Optional<Double> load = Optional.absent();
-      
+
       String endpointString = strEndpoints[i];
-      
-      matcher = endpointNamePattern.matcher(endpointString);
-      if(matcher.find()) {
-        endpoint = Optional.fromNullable(matcher.group(1));
+
+      for (Pattern endpointNameAllVersionsPattern : endpointNamePatterns) {
+        matcher = endpointNameAllVersionsPattern.matcher(endpointString);
+        if (matcher.find()) {
+          endpoint = Optional.fromNullable(matcher.group(1));
+          break;
+        }
       }
-      
-      matcher = endpointStatusPattern.matcher(endpointString);
-      if(matcher.find()) {
-        status = Optional.fromNullable(matcher.group(3) + " - " + simpleStates.getOrDefault("/" + endpoint.or(""), "UNKNOWN")) ;
+
+      for (Pattern endpointStatusPattern : endpointStatusPatterns) {
+        matcher = endpointStatusPattern.matcher(endpointString);
+        if (matcher.find()) {
+          status = Optional.fromNullable(matcher.group(3) + " - " + simpleStates.getOrDefault("/" + endpoint.or(""), "UNKNOWN"));
+          break;
+        }
       }
-      
-      matcher = endpointDcPattern.matcher(endpointString);
-      if(matcher.find()) {
-        dc = Optional.fromNullable(matcher.group(3));
+
+      for (Pattern endpointDcPattern : endpointDcPatterns) {
+        matcher = endpointDcPattern.matcher(endpointString);
+        if (matcher.find()) {
+          dc = Optional.fromNullable(matcher.group(3));
+          break;
+        }
       }
-      
-      matcher = endpointRackPattern.matcher(endpointString);
-      if(matcher.find()) {
-        rack = Optional.fromNullable(matcher.group(3));
+
+      for (Pattern endpointRackPattern : endpointRackPatterns) {
+        matcher = endpointRackPattern.matcher(endpointString);
+        if (matcher.find()) {
+          rack = Optional.fromNullable(matcher.group(3));
+          break;
+        }
       }
-      
-      matcher = endpointSeverityPattern.matcher(endpointString);
-      if(matcher.find()) {
-        severity = Optional.fromNullable(Double.parseDouble(matcher.group(3)));
+
+      for (Pattern endpointSeverityPattern : endpointSeverityPatterns) {
+        matcher = endpointSeverityPattern.matcher(endpointString);
+        if (matcher.find()) {
+          severity = Optional.fromNullable(Double.parseDouble(matcher.group(3)));
+          break;
+        }
       }
-      
-      matcher = endpointLoadPattern.matcher(endpointString);
-      if(matcher.find()) {
-        load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
-        totalLoad+=load.or(0.0);
+
+      for (Pattern endpointLoadPattern : endpointLoadPatterns) {
+        matcher = endpointLoadPattern.matcher(endpointString);
+        if (matcher.find()) {
+          load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
```

Comment:
it's not possible for `new BigDecimal(matcher.group(3)).doubleValue()` to be `null`.

(this will instead cause a NPE if `matcher.group(3)` is `null`)